schema/resolvers.js

```diff
@@ -146,21 +146,40 @@ const Resolvers = {
     toggleValidationRule: (_, args, ctx) =>
       ctx.repositories.Validation.toggleValidationRule(args.input),
     updateValidationRule: (_, args, ctx) =>
-      ctx.repositories.Validation.updateValidationRule(args.input)
+      ctx.repositories.Validation.updateValidationRule(args.input),
+    createMetadata: (root, args, ctx) =>
+      ctx.repositories.Metadata.insert(args.input),
+    updateMetadata: (_, args, ctx) => {
+      const typeValues = {
+        Date: "dateValue",
+        Text: "textValue",
+        Region: "regionValue",
+        Language: "languageValue"
+      };
+      const valueFieldName = typeValues[args.input.type];
+
+      return ctx.repositories.Metadata.update({
+        ...omit(args.input, valueFieldName),
+        value: args.input[valueFieldName]
+      });
+    },
```

Comment:
Could we move this into the repository in the interest of keeping the resolvers as simplistic as possible?